azure/target/blobstorage/blobStorageFunction/index.js

```diff
@@ -0,0 +1,244 @@
+ * @param LogRecords
+ */
+const fetchServiceName = (LogRecords) => {
+    if (!LogRecords.resourceId) {
+        error('Could not find resourceId for the log record');
+    }
+    var resourceInfo = LogRecords.resourceId.substring(1, LogRecords.resourceId.length);
+    var resourceArr = resourceInfo.split('/');
+    resourceArr.forEach(resource => {
+        if (resource.toUpperCase().startsWith("MICROSOFT.")) {
+            LogRecords.event_provider = "AZURE_" + resource.split('.')[1];
+        }
+    });
+    if(resourceArr[resourceArr.length - 2].toUpperCase()!=="PROVIDERS"){
+        LogRecords.eventsource = resourceArr[resourceArr.length - 2];
+    }
+};
+
+/**
+ *Processing and sending Log to vrli cloud for Event Hub.
+ * @param ServiceLogs
+ * @param collector
+ */
+const sendServiceLogsFromEventHub = (ServiceLogs, collector, trigger) => {
+    if (ServiceLogs[0].records) {
+        ServiceLogs.forEach((message) => {
+            message.records.forEach((record) => {
+                const data = collector.processLogsJsonFromEventHub(record);
+                collector.postDataToStream(data);
+            });
+        });
+    }
+};
+
+/**
+ * Processing and sending Log to vrli cloud for blob storage.
+ * @param serviceLogs
+ * @param collector
+ * @returns {Promise | Promise<any>}
+ */
+const sendServiceLogsFromBlobStorage = (serviceLogs, collector, trigger) => {
+    var logs;
+    if (typeof serviceLogs === 'string') {
+        logs = serviceLogs.trim().split('\n');
+    } else if (Buffer.isBuffer(serviceLogs)) {
+        logs = serviceLogs
+            .toString('utf8')
+            .trim()
+            .split('\n');
+    } else {
+        logs = JSON.stringify(serviceLogs)
+            .trim()
+            .split('\n');
+    }
+    logs.forEach(log => {
+        const data = collector.processLogsJsonFromBlobStorage(log);
+        return collector.postDataToStream(data);
+    });
+};
+
+class ServiceHttpCollector extends Collector {
+    constructor(lintEnv) {
+        super('simple', lintEnv);
+    }
+
+    //Processing Logs for eventhub
+    processLogsJsonFromEventHub(logsJson) {
+        if (!logsJson) {
+            throw new Error('JSON blob does not have log records. Skip processing the blob.');
+        }
+        const logSource = "event_hub";
+        processLogs(logsJson, logSource);
+        return JSON.stringify(logsJson);
+    }
+
+    //Processing logs for blob
+    processLogsJsonFromBlobStorage(logsJson) {
+        if (!logsJson) {
+            throw new Error('JSON blob does not have log records. Skip processing the blob.');
+        }
+        const logSource = "blob_storage";
+        logsJson = processLogTextAsJson(logsJson);
+        processLogs(logsJson, logSource);
+        return JSON.stringify(logsJson);
+    }
+}
+
+/**
+ * Error handeling.
+ * @param error
+ * @param context
+ */
+const handleError = (error, context) => {
+    context.fail(error);
+    console.log(error);
```

Comment:
just cross check if the log should be above context.fail()

log first and then fail

Reply:
Implemented review comments